configuration/azure/appconfig/appconfig.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package appconfig
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig"
+	"github.com/google/uuid"
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/dapr/components-contrib/configuration"
+	azauth "github.com/dapr/components-contrib/internal/authentication/azure"
+
+	"github.com/dapr/kit/logger"
+)
+
+const (
+	host                 = "appConfigHost"
+	connectionString     = "appConfigConnectionString"
+	maxRetries           = "maxRetries"
+	retryDelay           = "retryDelay"
+	maxRetryDelay        = "maxRetryDelay"
+	defaultMaxRetries    = 3
+	defaultRetryDelay    = time.Second * 4
+	defaultMaxRetryDelay = time.Second * 120
+)
+
+// ConfigurationStore is a Azure App Configuration store.
+type ConfigurationStore struct {
+	client   *azappconfig.Client
+	json     jsoniter.API
+	metadata metadata
+
+	logger logger.Logger
+}
+
+// NewAzureAppConfigurationStore returns a new Azure App Configuration store.
+func NewAzureAppConfigurationStore(logger logger.Logger) configuration.Store {
+	s := &ConfigurationStore{
+		json:   jsoniter.ConfigFastest,
+		logger: logger,
+	}
+
+	return s
+}
+
+// Init does metadata and connection parsing.
+func (r *ConfigurationStore) Init(metadata configuration.Metadata) error {
+	m, err := parseMetadata(metadata)
+	if err != nil {
+		return err
+	}
+	r.metadata = m
+
+	coreClientOpts := azcore.ClientOptions{
+		Telemetry: policy.TelemetryOptions{
+			ApplicationID: "dapr-" + logger.DaprVersion,
+		},
+	}
+
+	options := azappconfig.ClientOptions{
+		ClientOptions: coreClientOpts,
+	}
+
+	if r.metadata.connectionString != "" {
+		r.client, err = azappconfig.NewClientFromConnectionString(connectionString, &options)
+		if err != nil {
+			return err
+		}
+	} else {
+		var settings azauth.EnvironmentSettings
+		settings, err = azauth.NewEnvironmentSettings("appconfig", metadata.Properties)
+		if err != nil {
+			return err
+		}
+
+		var cred azcore.TokenCredential
+		cred, err = settings.GetTokenCredential()
+		if err != nil {
+			return err
+		}
+
+		r.client, err = azappconfig.NewClient(r.metadata.host, cred, &options)
+		if err != nil {
+			return err
+		}
+	}
+
+	return err
```

Comment:
Nit: if we're here, `err` is certainly `nil`, so I'd just do `return nil` to make it clear.